dashboard/app/controllers/policy_compliance_controller.rb

```diff
@@ -19,8 +22,8 @@ def child_account_consent
     #Get User
     user = permission_request.user
     #Update User
-    if user.child_account_compliance_state != User::ChildAccountCompliance::PERMISSION_GRANTED
-      user.child_account_compliance_state = User::ChildAccountCompliance::PERMISSION_GRANTED
+    if user.child_account_compliance_state != Policies::ChildAccount::ComplianceState::PERMISSION_GRANTED
+      user.child_account_compliance_state = Policies::ChildAccount::ComplianceState::PERMISSION_GRANTED
```

Comment:
I'm noticing here that we're directly checking and setting the property on the user, while I see elsewhere we check with `Policies::ChildAccount::compliant?(...)` and we update with `Services::ChildAccount.update_compliance(...)`

Feels inconsistent, though maybe I'm missing something.

Reply:
Kinda inconsistent, kinda not. One benefit of the `Services`, `Queries`, and `Policies` is that we can put feature specific code in an isolated module rather than a common file like `user.rb` or `application_controller.rb`. However, the code you pointed at here is in `policy_compliance_controller.rb` which only deals with the Child Account Policy feature. I have refactored the code you pointed out to `Services::ChildAccount` to test it out and I like the result. There is more code in this controller which could be refactored, but this PR is already pretty long and that code is unrelated to this particular JIRA item. I do think in the future we will refactor it thought.

Reply:
https://github.com/code-dot-org/code-dot-org/pull/52287/commits/63a47ee63b3c96654570eab9396dc58dbcae6f1d